docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -283,269 +725,152 @@ quarkus.oidc.credentials.secret=secret
 
-To support the integration with such OAuth2 servers, `quarkus-oidc` needs to be configured to allow the authorization code flow responses without `IdToken`: `quarkus.oidc.authentication.id-token-required=false`.
+For example, when working with GitHub, the Quarkus endpoint can acquire an `access` token which will allow it to request a GitHub profile of the current user.
 
-It is required because `quarkus-oidc` expects that not only `access` and `refresh` tokens but also `IdToken` will be returned once the authorization code flow completes.
+To support the integration with such OAuth2 servers, `quarkus-oidc` needs to be configured a bit differently to allow the authorization code flow responses without `IdToken`: `quarkus.oidc.authentication.id-token-required=false`.
 
-Note, even though you will configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` will be generated to support the way `quarkus-oidc` operates where an `IdToken` is used to support the authentication session and to avoid redirecting the user to the provider such as GitHub on every request. In this case the session lifespan is set to 5 minutes which can be extended further as described in the <<session-management,session management>> section.
+[NOTE]
+--
+Even though you will configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` will be generated to standardize the way `quarkus-oidc` operates: an `IdToken` is used to support the authentication session and to avoid redirecting the user to the provider such as GitHub on every request. In this case the session lifespan is set to 5 minutes which can be extended further as described in the <<session-management,session management>> section.
 
-The next step is to ensure that the returned access token can be useful to the current Quarkus endpoint.
-If the OAuth2 provider supports the introspection endpoint then you may be able to use this access token as a source of roles with `quarkus.oidc.roles.source=accesstoken`. If no introspection endpoint is available then at the very least it should be possible to request <<user-info,UserInfo>> from this provider with `quarkus.oidc.authentication.user-info-required` - this is the case with GitHub.
+This simplifies how you handle an application supporting multiple OIDC providers.
+--
 
-Configuring the endpoint to request <<user-info,UserInfo>> is the only way `quarkus-oidc` can be integrated with the providers such as GitHub.
+The next step is to ensure that the returned access token can be useful and is valid to the current Quarkus endpoint.
+The first way is to call the OAuth2 provider introspection endpoint by configuring `quarkus.oidc.introspection-path` if the provider offers such an endpoint.
+In this case you can use the access token as a source of role using `quarkus.oidc.roles.source=accesstoken`.
```

Comment:
Question: Not quite sure what is meant here by "as a source of role". Would it be correct to say something like" "In this case, you can specify `access token` as the value for the `quarkus.oidc.roles.source` property, that is, `quarkus.oidc.roles.source=access token`.

Reply:
@sheilamjones It should be a `a source of roles` instead of `a source of role`, thanks for highlighting it, I'd limit it though to this minor change only. For the code flow, the ID token typically contains roles, but there could be some variations (access token can be a source of this information, or userinfo)

Reply:
@sheilamjones I changed `role` to `roles` locally